### ISSUE TITLE
[Let's encrypt] cryptography.exceptions.InternalError: Unknown OpenSSL error

### ISSUE THREAD

**Issue by hillbicks** [pyca/cryptography#2619]
Hi guys,

while trying to install a letsencrypt cert I encountered the following error. 
```
./letsencrypt-auto certonly -a standalone
Updating letsencrypt and virtual environment dependencies...Traceback (most recent call last):
  File "/root/.local/share/letsencrypt/bin/pip", line 7, in <module>
    from pip import main
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/pip/__init__.py", line 15, in <module>
    from pip.vcs import git, mercurial, subversion, bazaar  # noqa
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/pip/vcs/mercurial.py", line 10, in <module>
    from pip.download import path_to_url
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/pip/download.py", line 38, in <module>
    from pip._vendor import requests, six
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/pip/_vendor/requests/__init__.py", line 53, in <module>
    from .packages.urllib3.contrib import pyopenssl
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/contrib/pyopenssl.py", line 53, in <module>
    import OpenSSL.SSL
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/OpenSSL/__init__.py", line 8, in <module>
    from OpenSSL import rand, crypto, SSL
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/OpenSSL/rand.py", line 11, in <module>
    from OpenSSL._util import (
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/OpenSSL/_util.py", line 6, in <module>
    from cryptography.hazmat.bindings.openssl.binding import Binding
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/cryptography/hazmat/bindings/openssl/binding.py", line 182, in <module>
    Binding.init_static_locks()
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/cryptography/hazmat/bindings/openssl/binding.py", line 139, in init_static_locks
    cls._ensure_ffi_initialized()
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/cryptography/hazmat/bindings/openssl/binding.py", line 134, in _ensure_ffi_initialized
    cls._register_osrandom_engine()
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/cryptography/hazmat/bindings/openssl/binding.py", line 99, in _register_osrandom_engine
    _openssl_assert(cls.lib, cls.lib.ERR_peek_error() == 0)
  File "/root/.local/share/letsencrypt/lib/python2.7/site-packages/cryptography/hazmat/bindings/openssl/binding.py", line 43, in _openssl_assert
    errors
cryptography.exceptions.InternalError: Unknown OpenSSL error. Please file an issue at https://github.com/pyca/cryptography/issues with information on how to reproduce this. ([_OpenSSLError(code=638025831L, lib=38, func=120, reason=103), _OpenSSLError(code=637964398L, lib=38, func=105, reason=110)])
```
it could very well be that I'm missing a package in my installation, so here is the list of installed pip packages:

```
pip list
cffi (1.4.2)
ClientForm (0.2.9)
cryptography (1.0.2)
enum34 (1.1.2)
feedparser (4.1)
idna (2.0)
iotop (0.6)
ipaddress (1.0.16)
M2Crypto (0.21.1)
mechanize (0.1.8)
ndg-httpsclient (0.4.0)
pip (7.1.2)
progressbar (2.2)
pyasn1 (0.1.9)
pycparser (2.14)
pycurl (7.19.0)
pygpgme (0.3)
pyOpenSSL (0.15.1)
pyxattr (0.5.0)
rssdler (0.4.0)
setuptools (0.6rc11)
six (1.10.0)
urlgrabber (3.9.1)
virtualenv (13.1.2)
wheel (0.26.0)
```
Hopefully someone can point me in the right direction.

Almost forgot: I already saw another issue with this error and it was solved for some people by installing cryptography 1.0.2. But that didn't do the trick for me.

Greets 
hillbicks

**Comment by reaperhulk** [comment]
@hillbicks could you try installing the latest cryptography (1.1.2 right now) rather than 1.0.2? I'm not sure how the lets-encrypt-auto package is triggering the import race condition you're seeing, but 1.1.2 does initialization significantly differently.

**Comment by hillbicks** [comment]
There is something about let's encrypt that is really strange and I think it is an issue on their end. 

I set up a new chroot build environment, installed everything via pip and got the same error. Funny is though, now nothing related to pip works anymore, not even pip list. It'll also result in the error I posted above. Unfortunately I'm not familiar enough with the inner workings of letsencrypt to guess what is going on here. 

So, unless you have an idea and what could be going on here, I guess I should open up a new issue on the letsencrypt repo.
